src/release/breaking-changes/android-14-nonlinear-text-scaling.md

```diff
@@ -0,0 +1,292 @@
+| `Text.textScaleFactor` getter                                           | 'textScaleFactor' is deprecated and shouldn't be used.        |
+| `EditableText({ double? TextScaleFactor = 1.0 })` constructor argument  | 'textScaleFactor' is deprecated and shouldn't be used.        |
+| `EditableText.textScaleFactor` getter                                   | 'textScaleFactor' is deprecated and shouldn't be used.        |
+
+### Material library
+
+| Affected APIs                                                                 | Error Message                                          |
+|-------------------------------------------------------------------------------|--------------------------------------------------------|
+| `SelectableText({ double? TextScaleFactor = 1.0 })` constructor argument      | 'textScaleFactor' is deprecated and shouldn't be used. |
+| `SelectableText.rich({ double? TextScaleFactor = 1.0 })` constructor argument | 'textScaleFactor' is deprecated and shouldn't be used. |
+| `SelectableText.textScaleFactor` getter                                       | 'textScaleFactor' is deprecated and shouldn't be used. |
+
+## Migration guide
+
+Widgets provided by the Flutter framework are already migrated. Migration is 
+needed only if you're using any of the deprecated symbols listed above. 
+
+### Migrating your APIs that expose `textScaleFactor`
+
+Before:
+```dart 
+abstract class _MyCustomPaintDelegate { 
+  void paint(PaintingContext context, Offset offset, double textScaleFactor) { 
+  }
+}
+```
+
+After:
+```dart 
+abstract class _MyCustomPaintDelegate { 
+  void paint(PaintingContext context, Offset offset, TextScaler textScaler) { 
+  }
+}
+```
+
+### Migrating code that consumes `textScaleFactor`
+
+If you're not currently using `textScaleFactor` directly, but rather passing it 
+to a different API that receives a `textScaleFactor`, and the receiver API has 
+already been migrated, then it's relateively straightforward:
+
+Before:
+```dart 
+RichText( 
+  textScaleFactor: MediaQuery.textScaleFactorOf(context),
+  ...
+)
+```
+
+After:
+```dart 
+RichText( 
+  textScaler: MediaQuery.textScalerOf(context),
+  ...
+)
+```
+
+If the receiver API hasn't been migrated yet, optionally, you could migrate your 
+code like so:
+
+Before:
+```dart 
+UnmigratedTextWidget( 
+  textScaleFactor: MediaQuery.textScaleFactorOf(context),
+  ...
+)
+```
+
+After:
+```dart 
+UnmigratedTextWidget( 
+  textScaleFactor: MediaQuery.textScalerOf(context).textScaleFactor,
+  ...
+)
+```
+
+If you wish to compute the scaled font size yourself, use `TextScaler.scale` 
+instead of the `*` binary operator:
+
+Before:
+```dart 
+final scaledFontSize = textStyle.fontSize * MediaQuery.textScaleFactorOf(context);
+```
+
+After:
+```dart 
+final scaledFontSize = MediaQuery.textScalerOf(context).scale(textStyle.fontSize);
+```
+
+
+If the API that provides `textScaleFactor` hasn't been migrated, consider 
```

Comment:
Basically move this before the "bad" migration advice above. (Or better: just remove that "bad" advice)